index.js

```diff
@@ -92,7 +92,11 @@ async function fastifyStatic (fastify, opts) {
     })
 
     wrap.on('pipe', function () {
-      reply.send(wrap)
+      if (request.method !== 'HEAD') { reply.send(wrap) }
+    })
+
+    wrap.on('finish', function () {
+      if (request.method === 'HEAD') { reply.send() }
```

Comment:
I would move this in the event handler above. I think this would not work for big files.

Reply:
@mcollina I tried to do this but the `pipe` event is not being triggered when using the HEAD route.

Reply:
Oh great!

Reply:
Then we can move the `if (request.method === 'HEAD')` out of the two event handler and only add the one that is needed.

Reply:
Anyway, I'm landing this now.